pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -2072,40 +2077,51 @@ func (cf *changeFrontier) advancePerTableProtectedTimestampRecord(
 	return true, nil
 }
 
-func (cf *changeFrontier) createPerTableProtectedTimestampRecord(
+func (cf *changeFrontier) createPerTableProtectedTimestampRecords(
 	ctx context.Context,
 	txn isql.Txn,
 	ptsEntries *cdcprogresspb.ProtectedTimestampRecords,
-	tableID descpb.ID,
-	tableHighWater hlc.Timestamp,
+	tableIDsToCreate map[descpb.ID]hlc.Timestamp,
 	pts protectedts.Storage,
 ) error {
-	// If the table is lagging and doesn't have a per table PTS record,
-	// we create a new one.
-	targets := changefeedbase.Targets{}
-	if cf.targets.Size > 0 {
-		err := cf.targets.EachTarget(func(target changefeedbase.Target) error {
-			if target.DescID == tableID {
-				targets.Add(target)
-			}
-			return nil
-		})
+	if ptsEntries.ProtectedTimestampRecords == nil {
+		ptsEntries.ProtectedTimestampRecords = make(map[descpb.ID]*uuid.UUID)
+	}
+	for tableID, tableHighWater := range tableIDsToCreate {
+		targets, err := cf.createPerTablePTSTarget(tableID)
 		if err != nil {
 			return err
 		}
+		ptr := createProtectedTimestampRecord(
+			ctx, cf.FlowCtx.Codec(), cf.spec.JobID, targets, tableHighWater,
+		)
+		uuid := ptr.ID.GetUUID()
+		ptsEntries.ProtectedTimestampRecords[tableID] = &uuid
+		if err := pts.Protect(ctx, ptr); err != nil {
+			return err
+		}
 	}
-	ptr := createProtectedTimestampRecord(
-		ctx, cf.FlowCtx.Codec(), cf.spec.JobID, targets, tableHighWater,
-	)
-	if ptsEntries.ProtectedTimestampRecords == nil {
-		ptsEntries.ProtectedTimestampRecords = make(map[descpb.ID]*uuid.UUID)
+	return writeChangefeedJobInfo(ctx, perTableProtectedTimestampsFilename, ptsEntries, txn, cf.spec.JobID)
+}
+
+func (cf *changeFrontier) createPerTablePTSTarget(
+	tableID descpb.ID,
+) (changefeedbase.Targets, error) {
+	targets := changefeedbase.Targets{}
+	if cf.targets.Size > 0 {
```

Comment:
Is this if condition necessary? Should we instead let this case return the `!found` assertion failure below?